prov/usnic/src/usdf_pep.c

```diff
@@ -336,6 +336,100 @@ usdf_pep_listen(struct fid_pep *fpep)
 	return 0;
 }
 
+/* Register as a callback triggered by the socket becoming writeable. Write as
+ * much data as can be written in a single write, and keep track of how much
+ * data is left. If the data is not fully written, it will finish getting
+ * written in another iteration of the progression.
+ */
+static int usdf_pep_reject_async(void *vreq)
+{
+	struct usdf_connreq *crp;
+	struct usdf_eq *eq;
+	int ret;
+
+	crp = vreq;
+	eq = crp->cr_pep->pep_eq;
+
+	ret = TEMP_FAILURE_RETRY(write(crp->cr_sockfd, crp->cr_ptr,
+				crp->cr_resid));
```

Comment:
Ick.  Though I can't imagine a situation where we would have usNIC on a non-glibc system, I find this glibc-specific macro to be... unpalatable.  I've never encountered it before today, necessitating a trip through google to the glibc online docs (after first futilely searching for it somewhere else in the libfabric code base).

You can keep it if you really want, but I would worry about getting bit by someone compiling with the wrong `feature.h` preprocessor definitions and getting a compilation failure.  The open coded version is quite readable, IMO.